src/parser/smt2/smt2.cpp

```diff
@@ -503,6 +500,12 @@ Command* Smt2::setLogic(std::string name, bool fromCommand)
   }
 
   d_logicSet = true;
+  if (isHOL())
```

Comment:
This has some issues.  One, `name` could already have `HO_` prefix.  

I would expect `isHOL` to force a call to `d_logicInfo.enableHigherOrder` immediately after `d_logic` is initialized, which I guess on the line 506 (master).

Reply:
The issue with that, and the source of the strange control flow it's bothering you here, is that `enableHigherOrder` resets the logic string. That is a common pattern in other `enable...` methods. Do you know why that is? Removing that things could be simplified.